diff --git a/RNWorklets.podspec b/RNWorklets.podspec
index fa66aa9ef6f6326e277e9890f2613c648f564361..0759598817fbaea50e1f036aa06aa0b1915e11f8 100644
--- a/RNWorklets.podspec
+++ b/RNWorklets.podspec
@@ -48,6 +48,9 @@ Pod::Spec.new do |s|
     s.dependency 'React-hermes'
   end
 
+  bundle_mode_flag = $worklets_config[:bundle_mode] ? 'WORKLETS_BUNDLE_MODE=1' : ''
+  hermes_v1_flag = ENV['RCT_HERMES_V1_ENABLED'] == '1' ? '-DHERMES_V1_ENABLED' : ''
+
   # React Native doesn't expose these flags, but not having them
   # can lead to runtime errors due to ABI mismatches.
   # There's also
@@ -55,8 +58,6 @@ Pod::Spec.new do |s|
   #   HERMESVM_PROFILER_BB
   # which shouldn't be defined in standard setups.
   hermes_debug_hidden_flags = 'HERMES_ENABLE_DEBUGGER=1'
-
-  bundle_mode_flag = $worklets_config[:bundle_mode] ? 'WORKLETS_BUNDLE_MODE=1' : ''
   
   s.pod_target_xcconfig = {
     "USE_HEADERMAP" => "YES",
@@ -75,7 +76,7 @@ Pod::Spec.new do |s|
     "CLANG_CXX_LANGUAGE_STANDARD" => "c++20",
     "GCC_PREPROCESSOR_DEFINITIONS[config=*Debug*]" => "$(inherited) #{hermes_debug_hidden_flags} #{bundle_mode_flag}",
     "GCC_PREPROCESSOR_DEFINITIONS[config=*Release*]" => "$(inherited) #{bundle_mode_flag}",
-    "OTHER_CFLAGS" => "$(inherited) #{feature_flags} #{version_flags}",
+    "OTHER_CFLAGS" => "$(inherited) #{feature_flags} #{version_flags} #{bundle_mode_flag} #{hermes_v1_flag}",
   }
   s.xcconfig = {
     "HEADER_SEARCH_PATHS" => [
